katsdpimageutils/primary_beam_correction.py

```diff
@@ -96,54 +108,50 @@ def central_freq(path):
     return np.array(c_freq_plane)
 
 
-def cosine_power_pattern(separation_rad, c_freq):
+def cosine_power_pattern(x, y, path, c_freq):
     """Compute Power patterns for a given frequency.
 
-    This uses the Cosine-squared power approximation from
-    Mauch et al. (2020).
+    This uses the katbeam module.
+    https://github.com/ska-sa/katbeam.git
 
     Parameters
     ----------
-    separation_rad : numpy array
-        Radial separation array
+    x, y : arrays of float of the same shape
+        Coordinates where beam is sampled, in degrees
+    path : str
+        FITS file
     c_freq : numpy array
         An array of central frequencies for each frequency plane
     """
-    rho = separation_rad
-    v_beam_rad = np.deg2rad(89.5 / 60.)
-    h_beam_rad = np.deg2rad(86.2 / 60.)
-    # Take the Geometric mean for the vertical and horizontal cut through the beam.
-    vh_beam_mean = np.sqrt(v_beam_rad * h_beam_rad)
+    # read in fits image and check the band type
+    raw_image = read_fits(path)
+    if round(raw_image.header['CRVAL3']/1e6) == 1284:  # L-band
+        circbeam = CircularBeam('MKAT-AA-L-JIM-2020')
+    if round(raw_image.header['CRVAL3']/1e6) == 816:  # UHF-band
+        circbeam = CircularBeam('MKAT-AA-UHF-JIM-2020')
     flux_density = []
     for nu in c_freq:
-        # Convert GHz to Hz
-        nu = nu/1.e9
-        theta_b = vh_beam_mean / nu
-        ratio = rho/theta_b
-        num = np.cos(1.189 * np.pi * ratio)
-        den = 1-4 * (1.189 * ratio)**2
-        a_b = (num/den)**2
-        flux_density.append(a_b)
+        nu = nu/1.e6  # GHz to MHz
+        a_b = circbeam.I(x, y, nu)
```

Comment:
This could just be called as I mentioned above.

Reply:
I have addressed it